From daec2151da57973f0dc17146c0058890d320c7dd Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 9 Aug 2018 15:54:29 -0500
Subject: [PATCH 4369/5242] MLK-18788: mtd: gpmi-nand: release HIGH_FREQ
 request in error path

commit  00c862ff400291ae6098b76d66bae80c0b707cef from
https://source.codeaurora.org/external/imx/linux-imx.git

release the HIGH_FREQ request was done in runtime suspend, disable
autosuspend and disable the runtime in NAND probe error path

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
index bb52f67..2c62471 100644
--- a/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
+++ b/drivers/mtd/nand/raw/gpmi-nand/gpmi-nand.c
@@ -2323,16 +2323,19 @@ static int gpmi_nand_probe(struct platform_device *pdev)
 
 	ret = gpmi_init(this);
 	if (ret)
-		goto exit_nfc_init;
+		goto exit_rpm;
 
 	ret = gpmi_nand_init(this);
 	if (ret)
-		goto exit_nfc_init;
+		goto exit_rpm;
 
 	dev_info(this->dev, "driver registered.\n");
 
 	return 0;
 
+exit_rpm:
+	pm_runtime_dont_use_autosuspend(this->dev);
+	pm_runtime_disable(this->dev);
 exit_nfc_init:
 	release_resources(this);
 exit_acquire_resources:
@@ -2344,6 +2347,7 @@ static int gpmi_nand_remove(struct platform_device *pdev)
 {
 	struct gpmi_nand_data *this = platform_get_drvdata(pdev);
 
+	release_bus_freq(BUS_FREQ_HIGH);
 	nand_release(nand_to_mtd(&this->nand));
 	gpmi_free_dma_buffer(this);
 	pm_runtime_disable(this->dev);
-- 
1.7.9.5

